frontend/templates/product/MetaTags.tsx

```diff
@@ -0,0 +1,66 @@
+import { useAppContext } from '@ifixit/app';
+import { Product, ProductVariant } from '@models/product';
+import Head from 'next/head';
+import React from 'react';
+
+export interface MetaTagsProps {
+   product: Product;
+   selectedVariant: ProductVariant;
+}
+
+export function MetaTags({ product, selectedVariant }: MetaTagsProps) {
+   const appContext = useAppContext();
+   const { metaTitle, shortDescription } = product;
+
+   const canonicalUrl = `${appContext.ifixitOrigin}/products/${product.handle}`;
+
+   const genericImages = React.useMemo(() => {
+      return product.images.filter((image) => {
+         return (
+            image.altText == null ||
+            !product.variants.find((variant) => variant.sku === image.altText)
+         );
+      });
+   }, [product.images, product.variants]);
+
+   const selectedVariantImages = React.useMemo(() => {
+      return product.images.filter((image) => {
+         const variant = product.variants.find(
+            (variant) => variant.sku === image.altText
+         );
+         return image.altText != null && variant?.id === selectedVariant.id;
+      });
+   }, [product.images, product.variants, selectedVariant.id]);
+
+   return (
+      <Head>
+         {metaTitle && (
+            <>
+               <title>{metaTitle}</title>
+               <meta property="og:title" content={metaTitle} />
+            </>
+         )}
+         {shortDescription && (
+            <>
+               <meta name="description" content={shortDescription} />
+               <meta name="og:description" content={shortDescription} />
+            </>
+         )}
+
+         <meta property="og:type" content="website" />
+
+         <link rel="canonical" href={canonicalUrl} />
+         <meta property="og:url" content={canonicalUrl} />
+
+         {genericImages.length > 0 &&
+            genericImages.map((image) => (
+               <meta key={image.id} property="og:image" content={image.url} />
+            ))}
+
+         {selectedVariantImages.length > 0 &&
+            selectedVariantImages.map((image) => (
+               <meta key={image.id} property="og:image" content={image.url} />
+            ))}
```

Comment:
> the Next Head component is not taking particular care of preserving the tag order

Is it possible, then, that the variant images could be ordered before the generic images? Is that a problem?

Reply:
Sure, likely and possible, but I don't want to consider that important unless we have evidence that it is.

Reply:
In this particular case it should not happen though. 
The reason is that on first render I've never noticed any out of order content. The out of order content seems to be generated only by client side updates of existing tags which are enqueued to the existing content.

Client side updates are only induced by a change in the currently selected variant.
Given that the generic images are memoized and depends only on the product, `og:image` tags associated with them should never be updated and should always precede the ones associated with the selected variant